src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -70,6 +74,13 @@ public async Task RunAsync(ActionRunStage stage)
                     // Set GITHUB_ACTION_PATH
                     step.ExecutionContext.SetGitHubContext("action_path", ActionDirectory);
 
+                    // Add Expression Functions
+                    step.ExecutionContext.ExpressionFunctions.Add(new FunctionInfo<AlwaysFunction>(PipelineTemplateConstants.Always, 0, 0));
```

Comment:
I need to test this but these could end up significantly more involved then expected. What if the composite action only runs on success and a step inside it only runs on failure? Might be good to only support conditionals based on context (and `Success` the default)

Reply:
good to scrutinize this more... i think there is some sensitivity when they should be added

Reply:
You are correct, I think this needs to be moved down to right before each individual composite step is invoked, let me do some testing